src/components/tl/ucp/reduce/reduce_knomial.c

```diff
@@ -141,7 +147,7 @@ ucc_status_t ucc_tl_ucp_reduce_knomial_start(ucc_coll_task_t *coll_task)
         args->src.info.buffer = args->dst.info.buffer;
     }
 
-    if (isleaf) {
+    if (isleaf && !self_avg) {
```

Comment:
maybe we move that also to init, this way we just set the scratch once.

Reply:
in the condition above dst is assigned to src and here src is assigned to scratch, so IMHO it is more readable to keep this here.